rxnetty/src/main/java/io/reactivex/netty/client/ConnectionPoolImpl.java

```diff
@@ -210,17 +250,64 @@ public void call(final Subscriber<? super ObservableConnection<I, O>> subscriber
 
     @Override
     public void shutdown() {
-        if (!isShutdown.compareAndSet(false, true)) {
+        if (!isShutdownRequested.compareAndSet(false, true)) {
             return;
         }
+        boolean shutdownPerformed = performShutdownIfPossible();
+        if (!shutdownPerformed) {
+            ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor(new ThreadFactory() {
```

Comment:
I changed this to Observable. Seems to be the better solution. Please have a look at it.